v2/workloadapi/client_test.go

```diff
@@ -1 +1,400 @@
 package workloadapi
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/spiffe/go-spiffe/v2/bundle/jwtbundle"
+	"github.com/spiffe/go-spiffe/v2/internal/test"
+	"github.com/spiffe/go-spiffe/v2/internal/test/fakeworkloadapi"
+	"github.com/spiffe/go-spiffe/v2/proto/spiffe/workload"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/go-spiffe/v2/svid/jwtsvid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestFetchX509SVID(t *testing.T) {
+	ca := test.NewCA(t)
+	wl := fakeworkloadapi.NewWorkloadAPI(t)
+	defer wl.Stop()
+	c, _ := New(context.TODO(), WithAddr(wl.Addr()))
+	defer c.Close()
```

Comment:
nit: `context.Background()` is more appropriate to uses in tests. From the docs:
> use context.TODO when it's unclear which Context to use or it is not yet available

Using `TODO` outside of that condition muddies up results when you grep for `TODO`s in the codebase when looking for this to fix up.

Also, we should be checking the error coming back from New.

```suggestion
	c, err := New(context.TODO(), WithAddr(wl.Addr()))
	require.NoError(t, err)	
	defer c.Close()
```